polaris-core/src/main/java/org/apache/polaris/core/context/RealmScope.java

```diff
@@ -16,12 +16,14 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.polaris.service.config;
+package org.apache.polaris.core.context;
 
-import org.apache.polaris.core.PolarisConfigurationStore;
+import jakarta.inject.Scope;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
 
-/** Interface allows injection of a {@link PolarisConfigurationStore} */
-public interface ConfigurationStoreAware {
-
-  void setConfigurationStore(PolarisConfigurationStore configurationStore);
-}
+@Scope
+@Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+public @interface RealmScope {}
```

Comment:
```suggestion
public @interface RealmScope extends AlterableContext {}
```

And have functionality to perform cleanup when a realm is removed.

Reply:
hmm, that's not in the jakarta cdi api jar